libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -120,6 +120,30 @@ uint32_t AP_ESC_Telem::get_active_esc_mask() const {
     return ret;
 }
 
+// return an active ESC for the purposes of reporting (e.g. in the OSD)
+uint8_t AP_ESC_Telem::get_active_esc() const
+{
+    uint32_t ret = 0;
+    float max_rpm = 0;
+    const uint32_t now = AP_HAL::millis();
+    uint32_t now_us = AP_HAL::micros();
+    for (uint8_t i = 0; i < ESC_TELEM_MAX_ESCS; i++) {
+        if (_telem_data[i].last_update_ms == 0 && !was_rpm_data_ever_reported(_rpm_data[i])) {
+            // have never seen telem from this ESC
+            continue;
+        }
+        if (_telem_data[i].stale(now)
+            && !rpm_data_within_timeout(_rpm_data[i], now_us, ESC_RPM_DATA_TIMEOUT_US)) {
+            continue;
+        }
```

Comment:
I think this is confusing. I would rather see that the ESC is not working than have it change to the next one and hide the issue.

Reply:
Maybe just two OSD items? "ESC max of all" which is what I think this code does? and add a new index param to the current ESC item with the default being to do whatever it is we do now. 

Another option might be to display the ESC number which the data is for alongside the data in the OSD.

Reply:
Fixed